gsl/span

```diff
@@ -630,7 +630,7 @@ template <class ElementType, std::ptrdiff_t Extent>
 constexpr ElementType& at(const span<ElementType, Extent>& s, size_t index)
 {
     // No bounds checking here because it is done in span::operator[] called below
-    return s[index];
+    return s[narrow_cast<typename span<ElementType, Extent>::index_type>(index)];
```

Comment:
I think you're fine to use `gsl::narrow_cast()` here because you know that `s` will validate the bounds anyway. So I like it as you have it.